static/js/components/tiles/top_event_tile.tsx

```diff
@@ -235,10 +241,74 @@ export function TopEventTile(props: TopEventTilePropType): JSX.Element {
             eventPlaces[event.placeDcid] = latLngPlaces[latLngKey];
           }
         });
-        setEventPlaces(eventPlaces);
+        return eventPlaces;
       })
       .catch(() => {
-        setEventPlaces({});
+        return {};
+      });
+  }
+
+  // Given a list of events, update eventPlaces with a map of event id to place
+  // that the event occurs within. Places will be of the same type as
+  // props.enclosedPlaceType.
+  function updateEventPlaces(events: DisasterEventPoint[]): void {
```

Comment:
ideally we'd do this fallback in flask so this can be done in a single request, or better yet, in the initial call to fetch event-data. could you add a todo here for now?

Reply:
done